File: LMSRepository/Models/LibraryAsset.cs
﻿using System;

namespace LMSLibrary.Models
{
    public class LibraryAsset
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public int Year { get; set; }
        public Status Status { get; set; }
        public decimal Cost { get; set; }
        public DateTime Added { get; set; }
        public int NumberOfCopies { get; set; }
        public int CopiesAvailable { get; set; }
        public string Description { get; set; }
        public AssetPhoto Photo { get; set; }
        public AssetType AssetType { get; set; }
        public Author Author { get; set; }
        public string ISBN { get; set; }
        public string DeweyIndex { get; set; }
        public Category Category { get; set; }
    }
}

File: src/spa/src/dto/models/library-asset.ts
/* tslint:disable */
/* eslint-disable */
import { AssetPhoto } from './asset-photo';
import { LibraryAssetAuthor } from './library-asset-author';
import { LibraryAssetCategory } from './library-asset-category';
import { LibraryAssetStatus } from './library-asset-status';
import { LibraryAssetType } from './library-asset-type';
export interface LibraryAsset {
  added?: string;
  assetAuthors?: null | Array<LibraryAssetAuthor>;
  assetCategories?: null | Array<LibraryAssetCategory>;
  assetType?: LibraryAssetType;
  copiesAvailable?: number;
  description?: null | string;
  deweyIndex?: null | string;
  id?: number;
  isbn?: null | string;
  numberOfCopies?: number;
  photo?: AssetPhoto;
  status?: LibraryAssetStatus;
  title?: null | string;
  year?: number;
}
